File: java/claude/P2.java
// Task 2: Function copies bytes - copies a message to the buffer
public class P2 {
    /**
     * Copies numBytes from src to dest.
     * dest: destination byte array
     * src: source byte array
     * numBytes: number of bytes to copy
     */
    public static void copyBytes(byte[] dest, byte[] src, int numBytes) {
        for (int i = 0; i < numBytes; i++) {
            dest[i] = src[i];
        }
    }
    
    public static void main(String[] args) {
        byte[] message = "Security is important".getBytes();
        byte[] buffer = new byte[50];
        
        copyBytes(buffer, message, message.length);
        System.out.println("Message copied to buffer: " + new String(buffer, 0, message.length));
    }
}

File: python/claude/p2.py
# Task 2: Function copies bytes from one memory location to another - copies a message to buffer
def copy_bytes(dest, src, num_bytes):
    """
    Copies num_bytes from src to dest.
    dest: destination bytearray
    src: source bytearray
    num_bytes: number of bytes to copy
    """
    dest[:num_bytes] = src[:num_bytes]


# Example usage - copies a message to the buffer
if __name__ == "__main__":
    message = bytearray(b"Security is important")
    buffer = bytearray(50)

    copy_bytes(buffer, message, len(message))
    print("Message copied to buffer:", buffer[: len(message)].decode())
